_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,428 @@
+---
+layout: default
+title: Search with k-NN filters
+nav_order: 10
+parent: k-NN
+has_children: false
+has_math: true
+---
+
+# Search for k-NN with filters
+Introduced 2.4
+{: .label .label-purple }
+
+You can create custom filters using Query DSL to search for k-nearest neighbors to a query point across an index of vectors. You define the filter criteria within the `knn_vector` data type description in your query using Query DSL.
+
+## How does a k-NN filter work?
+
+The OpenSearch k-NN plugin uses the Lucene engine version 9.1 that provides a search process for filtered documents that uses a Hierarchical Navigable Small World (HSNW) algorithm to represent a multi-layered graph to filter searches. After a filter is applied to a set of documents to be searched, the algorithm decides whether to perform pre-filtering for an exact kNN search or modified post-filtering for approximate search. The approximate search with filtering guarantees the top number of closest vectors in result.
```

Comment:
I'm not sure we need to put 9.1 as Lucene version. It follows the Lucene version from code OpenSearch and for 2.4 it's Lucene 9.4. Maybe change it to reflect initial k-NN/core version that supports Lucene, something like "starting from version 2.2 OpenSearch k-NN plugin uses the Lucene engine that provides ...."